pkg/workflow_execution.go

```diff
@@ -1028,3 +1092,96 @@ func (c *Client) SetWorkflowTemplateLabels(namespace, name, prefix string, keyVa
 
 	return filteredMap, nil
 }
+
+func (c *Client) GetWorkflowExecutionStatisticsForTemplate(workflowTemplate *WorkflowTemplate) (err error) {
+	tx, err := c.DB.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	query, args, err := sb.Select("name, created_at, finished_at").
+		From("workflow_executions").Where(sq.Eq{"workflow_template_id": workflowTemplate.ID}).OrderBy("created_at DESC").ToSql()
+	if err != nil {
+		return err
+	}
+	var workflowExecStats []WorkflowExecutionStatistic
+	err = c.DB.Select(&workflowExecStats, query, args...)
+
+	if err != nil {
+		return err
+	}
+
+	workflowTemplate.WorkflowExecutionStatisticReport = &WorkflowExecutionStatisticReport{}
+	if len(workflowExecStats) == 0 {
+		return
+	}
+
+	//Calculate and set the values
+	workflowTemplate.WorkflowExecutionStatisticReport.Total = int32(len(workflowExecStats))
+	createdAtTime := workflowExecStats[0].CreatedAt
+	workflowTemplate.WorkflowExecutionStatisticReport.LastExecuted = *createdAtTime
+	for _, workflowExecStat := range workflowExecStats {
+		if workflowExecStat.FailedAt != nil {
+			workflowTemplate.WorkflowExecutionStatisticReport.Failed++
+		}
+		if workflowExecStat.FinishedAt == nil {
+			workflowTemplate.WorkflowExecutionStatisticReport.Running++
+		}
+		if workflowExecStat.FinishedAt != nil {
+			workflowTemplate.WorkflowExecutionStatisticReport.Completed++
+		}
+	}
+	return
+}
+
+/**
+Will build a template that makes a CURL request to the onepanel-core API,
+with statistics about the workflow that was just executed.
+*/
+func GetExitHandlerWorkflowStatistics(client *Client, namespace string, workflowTemplateId *uint64) (workflowStepName, workflowStepTemplate, workflowStepWhen string, err error, wfv1Template wfv1.Template) {
+	workflowStepName = "workflow-statistics"
+	workflowStepTemplate = "workflow-statistics-template"
+	host := env.GetEnv("ONEPANEL_CORE_SERVICE_HOST", "")
+	if host == "" {
+		err = errors.New("ONEPANEL_CORE_SERVICE_HOST is empty.")
+		return
+	}
+	port := env.GetEnv("ONEPANEL_CORE_SERVICE_PORT", "")
+	if port == "" {
+		err = errors.New("ONEPANEL_CORE_SERVICE_PORT is empty.")
+		return
+	}
+	curlEndpoint := fmt.Sprintf("http://%s:%s/apis/v1beta1/%s/workflow_executions/{{workflow.name}}/statistics", host, port, namespace)
+
+	jsonRequestStruct := api.Statistics{
+		WorkflowStatus:     "{{workflow.status}}",
+		CreatedAt:          "{{workflow.creationTimestamp}}",
+		WorkflowTemplateId: int64(*workflowTemplateId),
+	}
+	jsonRequestBytes, err := json.Marshal(jsonRequestStruct)
+	if err != nil {
+		return "", "", "", err, wfv1.Template{}
+	}
+	jsonRequestStr := string(jsonRequestBytes)
+	curlJSONBody := fmt.Sprintf("--data '%s'", jsonRequestStr)
+
+	token, err := GetBearerToken(namespace)
+	if err != nil {
+		return "", "", "", err, wfv1.Template{}
+	}
+
+	wfv1Template = wfv1.Template{
+		Name: workflowStepTemplate,
+		Container: &corev1.Container{
+			Image:   "alpine:latest",
```

Comment:
Can't we use [the official curl image](https://hub.docker.com/r/curlimages/curl) so we don't have to do `apk add curl`?

Reply:
Good point, taking a look.

Reply:
Changed in feat/dev-core-144